### DISCUSSION TITLE
Should I buy an Electribe 2 for Hacktribe?

### DISCUSSION THREAD

**Discussion by testergrooves** [bangcorrupt/hacktribe#198] [Q&A] [answered]
I was so excited when they came out but was disappointed with the features. Back then I was lusting over an ESX-1SD with no idea how to use it and I was hoping Electribe 2 would be perfect but it isn't quite. Hacktribe sounds very exciting though!

There isn't too much documentation on E2 or HT so I have some basic questions to try and figure out if this device will work for my usecase. I just want to make demos on it using samples of vst instruments  and then will later use Ableton or Studio One to translate the resulting sound to midi and replace everything with VSTs so I don't care much about the sound or filters or FX, just the sequencing functionality grooves and ability to load samples and play them chromatically.

1. Is there no way to do pitchbend on the E2? I owned a Korg padKontrol before and always assumed the touchpad was primarily for pitch and mod but it seems to only be for MFX and touch scale. Having owned a padKontrol I consider touch scale useless especially with all those pads right there.

2. If I get an E2 and can upload my own grooves with Hacktribe I will literally put hundreds or thousands of grooves on it and use this feature more than anything else. Is that a problem? Will I run out of space? Can grooves be put in folders?

3. I won't easily be able to just add my midi grooves to the E2 since the grooves are python code, but is making grooves that only affect timing as easy as it seems? Just set a number for each of the 16 steps in a text file, upload it to the device, and I'm done and can upload it? And of course there's no way to efficiently translate these numbers from a midi file or Ableton groove or something right?

4. (Figured this one out)

5. I've heard that pattern chaining is terrible and there's like a skip between each pattern and some problem with pads staying muted or controls changing or something. Is it unusable for making a whole song on here?

6. I saw something about PCM on the E2 and how it doesn't work with Hacktribe. I don't really know what PCM is but in the Caustic mobile DAW that's the soundfont player. Can the E2 either stock or Hacktribe have sf2 files played on it?

7. Can you program triplet notes on the E2?

**Comment by BKLronin** [comment]
I personally would not buy a new electribe 2. 

While it is a good groovebox feature wise. There are some things that are a bit outdated, The sequencer especially isnt very flexbile, so you have to rely heavily on groove patterns and can´t play unquantized.

Every groovebox has its weakness by design of course but I have the feeling korg might release something in that regard maybe this year already. THe now got the sequencer right for the drumlogue so technically they have quita a few building blocks for a groovebox again. <- speculating

**Comment by bangcorrupt** [comment, marked as answer]
I can't tell you what to buy.  Assume there will be no new features, until they are already implemented. 

1. I'd like to add pitch bend at some point.  Also expand the use of the touch pad so it can map to arbitrary parameters.  You could map the pitch bend wheel to send a CC controlling oscillator pitch.

2. I've allocated space for 128 groove templates.  These can be loaded from a computer now, hopefully at some point from the SD card.  

3. At some point I'll write a script to build groove templates from MIDI files.  

4. Ok.

5. Pattern chaining works well.  I'd like to expand Event Recording into an editable linear sequencer at some point.  

6. PCM (Pulse Code Modulation) is a format for raw audio data.  I doubt I'll ever add sf2 support to Hacktribe, but it would be a good feature for a custom firmware.

7. Yes.

**Reply by javegon** [reply]
I have a couple of doubts -

Are there any CC map of all parameters ? or maybe easier than from Korg webpage. I've download the cc map... I can't understand it. I would like to use a midi controller in order to have more control, my idea would be to have different faders to control the volume of each one of the 16 parts, or maybe filter... I don't know if that is possible.

Event Recording - I think with pattern set it is possible use Event Recording, and have a kind of linear workflow, but I don't know if mute or other configurations are saved/kept between patterns, for example part 1 muted through different patterns.

PCM - I didn't know doesn't work at hack tribe. Well for me PCM mean a possibility of make a loop at the end of the sample. set star point . But maybe I am wrong... if so, could you show me a example of that in YouTube video? and what are the options and advantaged. Also what is the extension of PCM files? .pcm ? I can't find anything about PCM on E2S.

**Comment by egasimus** [comment]
Re #7: Maybe an alternate version of the 1/32 roller where the knob controls the ratchet duration (1/3, 1/4, 1/6, 1/12, 1/16, etc) instead of the depth?

**Comment by javegon** [comment]
@testergrooves For me E2 is a quite good sequencer + synth + sampler. There are a lot of machine out there, and each one has his own limitations and specific workflow. E2 has 16 parts, all of then with his midi channel, all parts are able to make synth, sampling sequencing, all able to play chromatically, all Paraphonic (a kind of polyphonic), all are able to make chords, all are able to setup at different  measures length (in order to make polyrhythmics patterns)  and all with slicing feature.  Some machines nowadays can't do those tasks IMO. 
About your question... work with an external sequencer, in order to transfer midi to Ableton would work fine, always under the limitations of each device, and also you should think about why you want to work remotely ... you have also midi controllers with step sequencers, arpeggios ... maybe is mobility ... are there any other smaller? this is quite small and portable with batteries. You will find clumsy and dumb the device in comparison with Ableton. @BKLronin told about unquantized patterns, well, in my case I don't need it. So it depends your needs. Anyway you have a groove patterns and swing, it is a kind of humanisation feature. And with imagination you can do everything you need, for example, if you have samples with different length of silent at the beginning of the sound you will have a unquantized patterns. it is a bit annoying and hard... but limitations is one of the thing I love to work with devices... make you  mind works.

About your questions
1 Pitchbend - you have a pitch button. Only this controller.   @bangcorrupt has mentioned about he is working to implement a pitchblend with touch pad. Another solutions is add some midi controller to the Electribe and map it with midi. IMO touch scale it really useful in order to make random melodies, random arp or work arp with step jump. It works really fine.
2 Grooves templates - 1000 of groove is too much my dude. I my opinion less is better than unlimited. Is better have 15 groove and know them perfectly than 1000 and don't have idea how them works. Also I don't really know if Electribe transfer midi with that groove applied, I would say no ... I've never tried. So, if your idea is compose music with that groove you should get that info before.
3 To have your own groove templates nowadays is imposible in this device. As @bangcorrupt has mentioned maybe in the next future.
5 Pattern chain. - Each pattern is a different song, if you know that, you can work well. I think every device work in the same way. The only problem IMO Is that the effects are affected by new pattern, for example, if you have a delay and you have a compresor in the next pattern the delay will be deleted abrupt . I usually keep the same controls between pattens and I don't usually have problems. 
6 PCM - In Electribe PCM means that you can work with samples, and make a loop at the end of the sound in order to replicate organic sounds like flute, violin, etc ...  
7 Triplets - Of course - You have different measures for Beat.  16, 32 , 8 trip, 16 trip. Any device can make it if you setup the sequencer at 12steps.  @testergrooves What kind of music do you produce?

I hope I have helped you

**Comment by BKLronin** [comment]
I worked with the electribe since the EM1, EMX, ESC for a few weeks than the E2 since it came out. I made a few of my greatest hits on them despite the limitations. But I have chaged and I urgendtly need fluent grooves and odd natural "badly" played timings. Its still a good machine kind of but quantization makes me feel trapped and the menu a little too. The sampler might be a bit better as one get rid of the slightly muddy(strangely) pcms. But then again I'm not a sampler guy as making librarys leave me always unsatisfied and I begin to dig for samples again and again. 

It depends, as always. It has quite some voices and the complete chain available per track is super powerful. Since I noodled the pcm instruments to death I mostly only do deep basslines but have a hard time for the melody part except sinesque pianos with deep mods.

The factory patterns have some strange magic in them the sound like fully mixed and mastered. This can be done with  a lot of filter and the right effect on the right track etc. 

Maybe I ve just overheard the sounds a bit after 20 years :D 

I´d wait, they probably put out some groovebox next but the drumlogue can also be some kind of substitution for it.

**Reply by javegon** [reply]
Mmmmm I don't bet for new drum machine in korg. They realised drumlogue, that is his new line of products nowadays. no more electribe I guess .

About natural, not quantized ... there are some solutions like have your own samples, using different steps for each part in order to have polyrhythmics patterns, groove and swing can be your better friends, also play with edit butom with your samples. It is a 300 euros machine, korg electribe is not a Electron, novation drums I think it has 3 microsteps per step or akai has unquantize options. but these machines doesnt have the same options than Electribe.

Patterns in Electribe sounds good because they are made by Mistabishi and other artist. And mistabishi knows Electribe pretty well. 

If you like unquantized songs check it out, I made with electribe 2s. 

https://drive.google.com/file/d/1BHSEwy5EAQ_9D1mzYm3hOTr4PGtWigAo/view?usp=share_link

**Reply by BKLronin** [reply]
Unquantized means, you get what you played. Sure there are 20 ways to mod existing steps but thats not the point. Thats why I like hardwar so much. Dsps have these really short latencies, is it sub 2ms less than 1? 

Its great at that price point maybe the most complete.

**Reply by javegon** [reply]
yes, it is true. I totally agree.  Unquantized is what you played out of the grid. :) 
I tried to find solutions for that , but it is not.

**Reply by egasimus** [reply]
Yeah the Electribe 2 factory patterns truly demonstrate the capabilities of the device in the best possible light. But I've found it difficult to get that level of detail in my own patterns, and also they don't really sound like any mainstream genre of music, for what it's worth. Overall I agree that built-in PCMs are a little lackluster, the sampling quality however is on point. Also if you play with all tracks' volume turned up high, you'll easily run out of headroom and not be able to hear some of the sounds, so you gotta dial everything down. Best to put a stereo compressor after it :) Which reminds me - having just 2 outs can be limiting but there isn't much that can be done about that - other than sequencing other gear.

I was already running a blue Electribe with the Sampler firmware before they closed the backdoor, and what I was sorely missing pattern chaining (which they released in a later update so it was not possible to have that). Hacktribe has that, and really opens the possibilities of the machine. Now I'm happily layering external synths on top of the custom samples and the new waveforms, and I'm finally getting somewhere with my dawless setup. Electribe has the best "flow" among the multiple sequencers and grooveboxes I tried since getting off Ableton a few years back, maybe AKAI MPC is more powerful but there are a lot of operations that you can't do while the patterns are playing on the MPC.

I'm currently reverse engineering the format in which Electribe 2 stores patterns, and from the looks of it I'm not sure how easily would it be to bring real off-grid recording without having to reimplement the entire sequencer. There's only a fixed amount of memory for each pattern and I guess that corresponds to a fixed amount of steps? So maybe adding a "step offset" to each step would be viable, but it would still be 16 steps per page. @bangcorrupt can confirm whether my reasoning is correct

**Reply by bangcorrupt** [reply]
> adding a "step offset" to each step would be viable, but it would still be 16 steps per page.

This is correct, it's how the timing element of groove templates works, up to half a step early or late.  The event recorder has free time sequencing with millisecond resolution.

There is a post on korgforums with the pattern format reversed.  At some point I'll add it to the definitions in [e2formats.py](https://github.com/bangcorrupt/hacktribe-editor/blob/main/e2_formats.py).

**Reply by javegon** [reply]
so... technically we have an unquantized method to record us playing. But only in a finished product we could say, because that file .e2sev is not editable and also is a linear system.
it is interesting :)

**Reply by egasimus** [reply]
This post? http://www.korgforums.com/forum/phpBB2/viewtopic.php?t=95368 SICK! Thanks dude, you da real MVP :star_struck:

**Reply by BKLronin** [reply]
I guess the problem is that you need another timer in a seperate thread that measures the offset. 
Currently they just let it fall into the step slots within a certain timing so it doenst have to be measured accurately. 
`if event.time > time_frame_Start step1 and event.time < time_frame_1:
        write.step1(note = c3, velocity = 87, gate = 45)`
So even if we got a place to save the offset it still needs to be measured. 

I guess the eventrecorder is different. They record a continues curve with one value in it.
